src/status.py

```diff
@@ -74,58 +79,226 @@ def write_new_status(session, status, status_level, value, national_or_sites="Na
+        f'<h2 style="color:#63BCAF;font-size:24px;">{"Data Providers"}</h2>',
         unsafe_allow_html=True,
     )
 
-    # Get database URLs
-    db_url = os.getenv("DB_URL", None)
-    db_url_sites = os.getenv("SITES_DB_URL", None)
+    # Add css for making metrics smaller
+    st.markdown(
+        """
+        <style>
+            .stMetric > div * {
+                font-size: 1.25rem !important;
+            }
+            .stMetric > label * {
+                font-size: 0.875rem !important;
+            }
+        </style>
+        """,
+        unsafe_allow_html=True,
+    )
+    # Fetch the latest status from the Status API
+    data_statuses = fetch_data_providers_status()
+
+    if data_statuses:
+        # Display the status of each data provider
+        for item in data_statuses:
+            provider = item["provider"]
+            source = item["source"]
+            status = item["status"]
+            msg = item["statusMessage"]
+            details = item.get("details", {})
+
+            emoji = "🟢" if status == "ok" else "🔴"
+
+            col1, col2, col3 = st.columns([3, 1, 2])
+
+            with col1:
+                st.metric(source, f"{emoji} {provider}")
+
+            with col2:
+                st.metric("Status", status)
+
+            with col3:
+                st.markdown(f"""<div style="font-size: 0.875rem">Message</div>""", unsafe_allow_html=True)
+                st.markdown(f"<div style='font-size: 0.875rem; color:#ccc; margin-top: 0.25rem;'>{msg if msg else '–'}</div>", unsafe_allow_html=True)
+
+            if provider == "EUMETSAT":
+                if details:
+                    col1, col2, col3, col4 = st.columns([3, 1, 1, 1])
+
+                    with col1:
+                        st.metric("Latest (UTC)", details.get("latestTimestamp", "-"))
+
+                    with col2:
+                        st.metric("Complete", details.get("complete", "-"))
+
+                    with col3:
+                        st.metric("Incomplete", details.get("incomplete", "-"))
+
+                    with col4:
+                        undelivered = details.get("undeliveredPlanned", 0) + details.get("undeliveredUnplanned", 0)
+                        st.metric("Undelivered", undelivered)
+                with st.expander("Show full delivery timeline"):
+                    extra = get_eumetsat_details()
+                    result_data = extra.get("details", {}).get("results", [])
+
+                    if len(result_data) >= 2:
+                        records = result_data[0]
+                        headers = result_data[1]
+                        df = pd.DataFrame(records, columns=headers)
+                        df["datetime"] = pd.to_datetime(df["datetime"])
+                        df = df.sort_values("datetime", ascending=False)
+
+                        # Add compact chart
+                        from datetime import datetime, timedelta
+                        import altair as alt
```

Comment:
move imports to the top of the function please

Reply:
Ah yep sorry that's sloppy copy/pasting from a test file from me there 😅